### ISSUE TITLE
Add the fold_all checkpoints as a release asset to the repository. 

### ISSUE THREAD

**Issue by ArthurBoschet** [axondeepseg/model_seg_generalist#21]
We should add the fold_all checkpoints in a release asset so we can download this "lighter" model version within ADS.

_Originally posted by @hermancollin in https://github.com/axondeepseg/model_seg_generalist/pull/17#pullrequestreview-1977892534_

**Comment by hermancollin** [comment]
@ArthurBoschet I saw you uploaded the best_checkpoints for the `fold_all` models. Is that a good idea? Because IIUC there is no validation set when you train with the whole dataset, so I'm not even sure what the "best checkpoint" represents

**Comment by hermancollin** [comment]
We could check how far the best checkpoint is from the last checkpoint by looking at the training curve. As per https://github.com/MIC-DKFZ/nnUNet/issues/29, the validation scores are meaningless because they are computed on the training data.

**Comment by ArthurBoschet** [comment]
![image](https://github.com/axondeepseg/model_seg_generalist/assets/65563348/f5be4b7b-4d78-4c3f-96ec-1ca8718c0afb)
@hermancollin Here is the learning curve. Just like you said, the best checkpoint refers to the best training pseudo dice score which ends up being the latest epoch. Nevertheless, I also included the `fold_all/checkpoint_final.pth` in `duke` for the sake of clarity.

**Comment by hermancollin** [comment]
@ArthurBoschet Ah great! I'll upload the final checkpoints to be safe but they should indeed be more or less identical to the best checkpoints.

**Comment by hermancollin** [comment]
Done in [r20240416](https://github.com/axondeepseg/model_seg_generalist/releases/tag/r20240416)
